core/templates/domain/skill/skill-rights.model.ts

```diff
@@ -22,8 +22,10 @@ export interface SkillRightsBackendDict {
 }
 
 export class SkillRights {
-  // The '_skillId' can be null for an interstitial SkillRights object.
-  // Check function 'createInterstitialSkillRights' for reference.
+  // The '_skillId' is set to null for the interstitial object which
+  // is to be used in the Skill Editor until the 'SkillRights' object
+  // has been loaded from the backend. Check function 'loadSkill' in
```

Comment:
As I mentioned in the previous comment, this doesn't suffice. You would need to explain the **conceptual reason** for why skill ID has to remain null, similar to the comment for _preferredSiteLanguageCode etc. If there is no good reason, then it may be necessary to refactor the implementation slightly to get rid of the need for the interstitial object. If this is not possible, you should explain clearly why it is impossible to do this.

Reply:
@seanlip 
The way the following comment explains, the interstitial object created is bound to the Skill Editor and will be retained till the time the Skill Editor is active and hence, it needs to have some initial values. When the actual object loads from the backend, the values of the interstitial object **itself** are updated and this update propagates through the Skill Editor.
https://github.com/oppia/oppia/blob/45bbb461593eb7b2f861b584e16bf6d9094986b7/core/templates/pages/skill-editor-page/services/skill-editor-state.service.ts#L166-L176

The `_skillId` will initially be **(does not remain)** `null` since it is not bound to an actual skill in the backend, it is just used as a placeholder for the Skill Editor until the skill from the backend loads and its values can be populated and used.

The same approach has been used for `ConceptCardObject`, `TopicObject` and `SubtopicObject`.

In conclusion:

We need to create a `SkillRights` and `Skill` object which is bound to the Skill Editor when it initially opens and these objects are retained. The value of `_skillId` of this object is `null` initially because the actual `_skillId` of the skill needs to be loaded from the backend. When it is loaded, the value of `_skillId` of the original object is updated.

Reply:
But at the point the skill editor is opened, wouldn't we already know the skill ID? For instance, wouldn't the skill ID already be in the URL? I feel like I might be missing something here.

Also, you mention that the actual skillId needs to be loaded from the backend. Why does this not apply for the other properties of the skill?

Reply:
yes, the `skillId` will be present and is used to load both the `Skill` and `SkillRights` object associated with that specific Id. 

Maybe I worded it wrong, the actual `Skill` and `SkillRights` object needs to be loaded from the backend and then we use the values of all the properties of these objects to populate the Skill Editor.

Since I mentioned the same approach being used for `Topic`, we initialize all the values to a default, not just the `skillId` when we initially open the editor. 
https://github.com/oppia/oppia/blob/45bbb461593eb7b2f861b584e16bf6d9094986b7/core/templates/domain/topic/topic-rights.model.ts#L86-L89

Reply:
The default values are used since we do not know the initial values associated with the Ids since the objects are being loaded from the backend.

Reply:
Perhaps I'm missing something, but why does SkillRights need a skillId in the first place, anyway?

I also don't see anything corresponding to this in the backend. So I'm wondering, what happens if we get rid of the skillId completely from here, and rely on the Skill object instead as the source of truth for that?

Reply:
`skillId` is used to get the `SkillsRight` object given its Id. I am not sure by what you mean by getting rid of the `skillId` property since then we won't be able to identify which skill has that rights. Were you referring to adding the `can_edit_skill_description` property to the `Skill` Object itself and getting rid of the `SkillRights`?

Reply:
No, I'm just a bit confused about why skillId (or this entire object) is needed at all, since we don't actually store rights for skills.

Let's look at the `can_edit_skill_description` property. Is it ever different for different skills on the page? If not, then there's no point in duplicating it for every single skill, right -- it should just be something in a generally-accessible service?

Reply:
I checked the permissions and you are right, it is not different for different skills on the page.

Reply:
Then let's get rid of this object altogether and move the "can edit skill description" into a service (either the existing one or a new one, whichever makes sense). Will that work? You can consult with @kevintab95 if needed.

Reply:
@vojtechjelinek suggested moving the `can_edit_skill_description`  property to `UserInfo`. 
https://github.com/oppia/oppia/blob/c714eeb320e0ebf9fe400403569428933658d3f1/core/templates/domain/user/user-info.model.ts#L31-L40
@kevintab95 what do you think about this approach?

Reply:
Will cover this file in a separate PR.